File: working hours register/level 1/Python/worker.py
class Worker:
    def __init__(self, worker_id: str, position: str, compensation: int):
        self.worker_id = worker_id
        self.position = position
        self.compensation = compensation
        self.total_minutes = 0
        self.is_inside = False
        self.last_entry_time = None

    def register(self, timestamp: int) -> str:
        if not self.is_inside:
            self.is_inside = True
            self.last_entry_time = timestamp
        else:
            self.is_inside = False
            self.total_minutes += timestamp - self.last_entry_time
            self.last_entry_time = None
        return "registered"

    def get_total_minutes(self) -> str:
        return str(self.total_minutes)

File: working hours register/Level 2/C#/worker.cs
// Worker.cs
public class Worker
{
    public string WorkerId { get; }
    public string Position { get; }
    public int Compensation { get; }
    public int TotalMinutes => _totalMinutes;

    private bool _isInside = false;
    private int? _lastEntryTime = null;
    private int _totalMinutes = 0;

    public Worker(string workerId, string position, int compensation)
    {
        WorkerId = workerId;
        Position = position;
        Compensation = compensation;
    }

    public string Register(int timestamp)
    {
        if (!_isInside)
        {
            _isInside = true;
            _lastEntryTime = timestamp;
        }
        else
        {
            if (_lastEntryTime.HasValue)
            {
                _totalMinutes += timestamp - _lastEntryTime.Value;
                _lastEntryTime = null;
            }
            _isInside = false;
        }

        return "registered";
    }

    public string GetTotalMinutes()
    {
        return _totalMinutes.ToString();
    }
}
